src/cdomain/value/cdomains/addressDomain.ml

```diff
@@ -238,10 +238,9 @@ struct
     else
       ID.top_of ik
 
+  (** @raise Not_found if set is empty. *)
   let type_of xs =
-    try Addr.type_of (choose xs)
-    with (* WTF? Returns TVoid when it is unknown and stuff??? *)
-    | _ -> voidType
+    Addr.type_of (choose xs) (* TODO: what if ambiguous type? what if chooses NullPtr but also contains Addr with proper type? *)
```

Comment:
Are we sure it's safe to remove the `void` fallback for `Not_found`?

Reply:
I just moved it to two call sites in a35dc82ac7bcdbff4bc18f4bb20d051c20f3a9b0, where this otherwise lead to crashes on regression tests. And even those are kind of dubious: why should we ever try to get the type of an empty points-to set?
Even so, the worst that can now happen is that there's an uncaught `Not_found` (if that ever happens), whose stacktrace is easy to see and the fallback can easily be added to such call sites. I wouldn't do that _en masse_ because it itself would cover up problems where we try to operate on empty points-to set when they should've already been handled.

Reply:
I am more worried about the `Addr.type_of` throwing then providing an empty points-to-set.

Reply:
I think it may be crucial for some of our real-world programs. I think we should do a run for those after merging this.

Reply:
Sure, I don't added call-site catching where it was needed for our test suite. If some `Not_found` or `Type_of_error` now slips through, it will be uncaught and crashes Goblint with a stacktrace, so it's quite easy to see which call of `AD.type_of` should have a fallback (or better decide what on the result).
So, yes, if they come up (on sv-benchmarks or our bench stuff), we can easily put in such specific catch.